findDependencies.py.file

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python
+
+import sys, re, os, json, gzip
+from argparse import ArgumentParser
+from glob import glob
+
+parser = ArgumentParser()
+parser.add_argument('-rel')
+parser.add_argument('-arch', dest = 'scramarch', default = os.environ.get("SCRAM_ARCH"))
+parser.add_argument('-scramroot', default = os.environ.get("SCRAMV1_ROOT"))
+args = parser.parse_args()
+
+rel = args.rel
+scramarch = args.scramarch
+scramroot = args.scramroot
+
+if scramroot is None:
+  scramroot = os.popen("sh -v scram arch 2>&1 |  grep 'SCRAMV1_ROOT=' | sed 's|;.*||;s|SCRAMV1_ROOT=||;s|\"||g' | sed -e \"s|'||g\"").read()
+
+global name, dir
+uses = {}
+usedby = {}
+directory = rel + "/tmp" # root dir for traversing
+
+def doexec():
+  global uses
+  getnext = 0
+  depname = ""
+  with open(name, 'r') as file:
+    for l in file:
+      l = l.rstrip('\n')
+      if re.search(r'^[^:]+ :\s*$', l): break
+      l = re.sub(r'\s*\\$', r'', l)
+      sp1 = l.split()
+      if len(sp1) == 0: continue
+      if len(sp1[0]) < 4: continue
+      sp2 = sp1[0].split('/')
+      tsp1 = ""
+      foundsrc = 0
+      for t in sp2:
+        if foundsrc == 1: tsp1 += "%s/" % t
+        if t == "src": foundsrc = 1
+      tsp1 = tsp1[:-1]
+      if tsp1 == "": continue
+      if getnext == 1:
+        depname = tsp1
+        getnext = 0
+      else:
+        getnext = 0
+        if re.search( r'^tmp\/', sp1[0]):
+          if re.search( r'(\.o|\/a\/xr+\.cc):$', sp1[0]):
+            getnext = 1
+        else:
+          if re.search( r'^src', sp1[0]):
+            if depname not in uses.keys():
+              uses[depname] = "%s " % tsp1
+            else: 
+              uses[depname] += "%s " % tsp1
+            if tsp1 not in usedby.keys():
```

Comment:
no need for .keys() [in python2 this is a copy of a list]

Reply:
Thanks! The execution time drastically decreased:
with .keys()        |  without .keys()
--------------------------------------
real   1.8625      |  real  0.9135
user    1.781      |  user 0.8349
sys	 0.0813      |  sys   0.0786